### PR TITLE
bugs:  419917, 419922, 

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Bug 419917:When builld fails in ant it shouldnt call report code coverage in ant, maven and gradle cc tasks
Fix: set errorActionPreference to stop for enable code coverage. Made a check for availability of coverage.xml before trying to publish. Gave a help message saying" build might have failed. Fix build to get code coverage" .

Bug 419922:[Ant CC] When there is a test in the build.xml and when we dont execute it , We throw the exception saying jacococexec1.exec not found.

Used try and catch to avoid build from failing. Gave an warning message saying "There might be no tests"

### CODE REVIEW

File: Tasks/ANTPreview/ant.ps1 [hunk 1]
```diff
@@ -74,7 +74,7 @@ $CCReportTask = "CodeCoverage_" +[guid]::NewGuid()
 if($isCoverageEnabled)
 {
    # Enable code coverage in build file
-   Enable-CodeCoverage -BuildTool 'Ant' -BuildFile $antBuildFile -CodeCoverageTool $codeCoverageTool -ClassFilter $classFilter -ClassFilesDirectory $classFilesDirectory -SourceDirectory $srcDirectory -SummaryFile $summaryFileName -ReportDirectory $reportDirectoryName -CCReportTask $CCReportTask
+   Enable-CodeCoverage -BuildTool 'Ant' -BuildFile $antBuildFile -CodeCoverageTool $codeCoverageTool -ClassFilter $classFilter -ClassFilesDirectory $classFilesDirectory -SourceDirectory $srcDirectory -SummaryFile $summaryFileName -ReportDirectory $reportDirectoryName -CCReportTask $CCReportTask -ErrorAction Stop
```
Comment (ajay-arora) [inline Tasks/ANTPreview/ant.ps1]: can you scan the rest of the file as well for similar change if applicable

File: Tasks/ANTPreview/ant.ps1 [hunk 2]
```diff
@@ -116,8 +109,26 @@ else
...
-   Publish-CodeCoverage -CodeCoverageTool $codeCoverageTool -SummaryFileLocation $summaryFile -ReportDirectory $reportDirectory -Context $distributedTaskContext    
+   # run report code coverage task which generates code coverage reports.
+   Write-Verbose "Collecting code coverage reports" -Verbose
+   try
+   {
+		Invoke-Ant -AntBuildFile $antBuildFile -Targets $CCReportTask 
+   }
+   catch
+   {
+		Write-Warning "Failed to collect code coverage. There might be no tests." -Verbose
```
Comment (ajay-arora) [inline Tasks/ANTPreview/ant.ps1]: can you test this scenario manually and see if this message comes ?

File: Tasks/ANTPreview/ant.ps1 [hunk 3]
```diff
@@ -116,8 +109,26 @@ else
...
+   }
+   
+   if(Test-Path $summaryFile)
+   {
+		Write-Verbose "Calling Publish-CodeCoverage" -Verbose
+		Publish-CodeCoverage -CodeCoverageTool $codeCoverageTool -SummaryFileLocation $summaryFile -ReportDirectory $reportDirectory -Context $distributedTaskContext    
+   }
+   else
+   {
+		Write-Warning "No code coverage found to publish." -Verbose
```
Comment (ajay-arora) [inline Tasks/ANTPreview/ant.ps1]: can we print summaryFile

File: Tasks/ANTPreview/ant.ps1 (hunk 1 above)
Comment (AshwiniChalla) [inline Tasks/ANTPreview/ant.ps1]: When ant fails --> either build failed or some test failed. Hence we should not stop there but publish whatever existing.Hence I felt this is the only place appropriate for this

File: Tasks/ANTPreview/ant.ps1 (hunk 2 above)
Comment (AshwiniChalla) [inline Tasks/ANTPreview/ant.ps1]: If you mean waarning. Yes i can see the warning when the report collection fails

File: Tasks/GradlePreview/Gradle.ps1 [hunk 4]
```diff
@@ -114,8 +114,15 @@ else
...
-   Write-Verbose "Calling Publish-CodeCoverage" -Verbose
-   Publish-CodeCoverage -CodeCoverageTool $codeCoverageTool -SummaryFileLocation $summaryFile -ReportDirectory $reportDirectory -Context $distributedTaskContext    
+	if(Test-Path $summaryFile)
+	{
+		Write-Verbose "Calling Publish-CodeCoverage" -Verbose
+		Publish-CodeCoverage -CodeCoverageTool $codeCoverageTool -SummaryFileLocation $summaryFile -ReportDirectory $reportDirectory -Context $distributedTaskContext   
+	}
+	else
+	{
+		Write-Warning "No code coverage found to publish." -Verbose
```
Comment (ajay-arora) [inline Tasks/GradlePreview/Gradle.ps1]: can we add some potential fixes here so as to make the error actionable

File: Tasks/ANTPreview/ant.ps1 [hunk 5]
```diff
@@ -1,4 +1,4 @@
-﻿param (
+param (
     [string]$antBuildFile,
     [string]$options,
```
Comment (ajay-arora) [inline Tasks/ANTPreview/ant.ps1]: looks fine overall

File: Tasks/ANTPreview/ant.ps1 (hunk 3 above)
Comment (AshwiniChalla) [inline Tasks/ANTPreview/ant.ps1]: Will do. Will print summaryFile and ReportDirectory

File: Tasks/GradlePreview/Gradle.ps1 (hunk 4 above)
Comment (AshwiniChalla) [inline Tasks/GradlePreview/Gradle.ps1]: Added

**Pull request merged**